pkg/sget/blob.go

```diff
@@ -0,0 +1,214 @@
+	if err != nil {
+		return err
+	}
+
+	certs, err := extractCerts(tlogEntry)
+	if err != nil {
+		return err
+	}
+	cert = certs[0]
+	pubKey, err = sigstoresigs.LoadECDSAVerifier(cert.PublicKey.(*ecdsa.PublicKey), crypto.SHA256)
+	if err != nil {
+		return err
+	}
+
+	if err := pubKey.VerifySignature(bytes.NewReader(targetSig), bytes.NewReader(artifact)); err != nil {
+		return err
+	}
+
+	if cert != nil { // cert
+		if err := cosign.TrustedCert(cert, fulcio.GetRoots()); err != nil {
+			return err
+		}
+		fmt.Fprintln(os.Stderr, "Certificate is trusted by Fulcio Root CA")
+		fmt.Fprintln(os.Stderr, "Email:", cert.EmailAddresses)
+		for _, uri := range cert.URIs {
+			fmt.Fprintf(os.Stderr, "URI: %s://%s%s\n", uri.Scheme, uri.Host, uri.Path)
+		}
+		fmt.Fprintln(os.Stderr, "Issuer: ", sigs.CertIssuerExtension(cert))
+	}
+	fmt.Fprintln(os.Stderr, "Verified OK")
+
+	rekorClient, err := rekor.NewClient(sg.RekorURL)
+	if err != nil {
+		return err
+	}
+	var pubBytes []byte
+
+	pubBytes, err = sigs.PublicKeyPem(pubKey, signatureoptions.WithContext(ctx))
+	if err != nil {
+		return err
+	}
+
+	if cert != nil {
+		pubBytes, err = cryptoutils.MarshalCertificateToPEM(cert)
+		if err != nil {
+			return err
+		}
+	}
+
+	b64sig := base64.StdEncoding.EncodeToString(targetSig)
+	if err != nil {
+		return err
+	}
+
+	uuid, index, err := cosign.FindTlogEntry(ctx, rekorClient, b64sig, artifact, pubBytes)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(os.Stderr, "tlog entry verified with uuid: %q index: %d\n", uuid, index)
+
+	_, err = sg.Out.Write(artifact)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO(shibumi): we might want to export this function or add this to our package?!
+func extractCerts(e *models.LogEntryAnon) ([]*x509.Certificate, error) {
+	b, err := base64.StdEncoding.DecodeString(e.Body.(string))
+	if err != nil {
+		return nil, err
+	}
+
+	pe, err := models.UnmarshalProposedEntry(bytes.NewReader(b), runtime.JSONConsumer())
+	if err != nil {
+		return nil, err
+	}
+
+	eimpl, err := types.NewEntry(pe)
+	if err != nil {
+		return nil, err
+	}
+
+	var publicKeyB64 []byte
+	switch e := eimpl.(type) {
+	case *rekord.V001Entry:
+		publicKeyB64, err = e.RekordObj.Signature.PublicKey.Content.MarshalText()
+		if err != nil {
+			return nil, err
+		}
+	case *hashedrekord.V001Entry:
+		publicKeyB64, err = e.HashedRekordObj.Signature.PublicKey.Content.MarshalText()
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, errors.New("unexpected tlog entry type")
+	}
+
+	publicKey, err := base64.StdEncoding.DecodeString(string(publicKeyB64))
+	if err != nil {
+		return nil, err
+	}
+
+	certs, err := cryptoutils.UnmarshalCertificatesFromPEM(publicKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(certs) == 0 {
+		return nil, errors.New("no certs found in pem tlog")
+	}
+
+	return certs, err
+}
+
+// TODO(shibumi): we might want to export this function or add this to our package?!
+func isb64(data []byte) bool {
```

Comment:
Does it make sense to export this function and add it to another package and then re-import the function here?
Right now we are having the code twice (once in the sget code and once in the cosign code).